libraries/standard/mqtt/src/iot_mqtt_validate.c

```diff
@@ -600,7 +600,8 @@ bool _IotMqtt_ValidateConnect( const IotMqttConnectInfo_t * pConnectInfo )
         /* If will info is provided, check that it is valid. */
         if( pConnectInfo->pWillInfo != NULL )
         {
-            status = _IotMqtt_ValidateLwtPublish( pConnectInfo->awsIotMqttMode,
+            status = pConnectInfo->pWillInfo->pPayload != NULL &&
+                    _IotMqtt_ValidateLwtPublish( pConnectInfo->awsIotMqttMode,
```

Comment:
Looks from the description that we are trying to avoid incorrect parameters but we are still makng the Validate publish call even when we think the data is wrong?

Also here it seems the Validate function is supposed to return some result which we should be checking? Are we not doubling up on the check now?

Reply:
Instead of anding the result from publish into the result, I could: 
```c
if(pConnectInfo->pWillInfo->pPayload != NULL)
{
    status = _IotMqtt_ValidateLwtPublish( pConnectInfo->awsIotMqttMode,
                                          pConnectInfo->pWillInfo );
}
```
We should not update the `_validatePublish`, because pPayload != NULL is specific to ConnectInfo. There are other functions that might accept pPayload == NULL iff payloadLength == 0.

Reply:
Ok makes sense. Can we please just put brackets in that status asignment as it will confuse people.

Reply:
Done.